### PR TITLE
Federated monitoring basic auth

### PR DESCRIPTION
**Pull request by tarokkk**

### CODE REVIEW

File: internal/monitor/cluster_subscriber.go [hunk 1]
```diff
@@ -112,44 +121,44 @@ func (s *clusterSubscriber) AddClusterToPrometheusConfig(clusterID uint) {
...
-		))
-
+		s.errorHandler.Handle(err)
 		return
 	}
-	config, err := k8sclient.NewClientConfig(kubeConfig)
-	if err != nil {
-		s.errorHandler.Handle(err)
-
+	if len(secrets) < 1 {
```
Comment (orymate) [inline internal/monitor/cluster_subscriber.go]:
```suggestion
	if len(secrets) == 0 {
```

File: internal/monitor/cluster_subscriber.go [hunk 2]
```diff
@@ -184,14 +193,14 @@ func (s *clusterSubscriber) RemoveClusterFromPrometheusConfig(orgID uint, cluste
...
-	delete(secret.StringData, fmt.Sprintf("%s_%s_client-key-data.pem", org.Name, clusterName))
-	delete(secret.Data, fmt.Sprintf("%s_%s_certificate-authority-data.pem", org.Name, clusterName))
-	delete(secret.Data, fmt.Sprintf("%s_%s_client-certificate-data.pem", org.Name, clusterName))
-	delete(secret.Data, fmt.Sprintf("%s_%s_client-key-data.pem", org.Name, clusterName))
+	//delete(secret.StringData, fmt.Sprintf("%s_%s_certificate-authority-data.pem", org.Name, clusterName))
+	//delete(secret.StringData, fmt.Sprintf("%s_%s_client-certificate-data.pem", org.Name, clusterName))
+	//delete(secret.StringData, fmt.Sprintf("%s_%s_client-key-data.pem", org.Name, clusterName))
+	//delete(secret.Data, fmt.Sprintf("%s_%s_certificate-authority-data.pem", org.Name, clusterName))
+	//delete(secret.Data, fmt.Sprintf("%s_%s_client-certificate-data.pem", org.Name, clusterName))
+	//delete(secret.Data, fmt.Sprintf("%s_%s_client-key-data.pem", org.Name, clusterName))
```
Comment (orymate) [inline internal/monitor/cluster_subscriber.go]: remove these

File: internal/monitor/cluster_subscriber.go [hunk 3]
```diff
@@ -112,44 +121,44 @@ func (s *clusterSubscriber) AddClusterToPrometheusConfig(clusterID uint) {
...
-		orgName:        org.Name,
-		clusterName:    c.GetName(),
-		endpoint:       apiEndpoint,
-		caCertFileName: fmt.Sprintf("%s_%s_certificate-authority-data.pem", org.Name, c.GetName()),
-		certFileName:   fmt.Sprintf("%s_%s_client-certificate-data.pem", org.Name, c.GetName()),
-		keyFileName:    fmt.Sprintf("%s_%s_client-key-data.pem", org.Name, c.GetName()),
+	query := &pkgSecret.ListSecretsQuery{
+		Type: pkgSecret.TLSSecretType,
+		Tags: []string{
+			fmt.Sprintf("clusterUID:%s", c.GetUID()),
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply internal/monitor/cluster_subscriber.go]: Cluster UID is UUID -> string

File: cluster/hooks.go [hunk 4]
```diff
@@ -216,6 +256,16 @@ func InstallMonitoring(input interface{}) error {
...
 				"affinity":    getHeadNodeAffinity(cluster),
 				"tolerations": getHeadNodeTolerations(),
+				"ingress": map[string]interface{}{
+					"enabled": true,
+					"annotations": map[string]string{
+						"traefik.ingress.kubernetes.io/auth-type":   "basic",
+						"traefik.ingress.kubernetes.io/auth-secret": prometheusSecretName,
+					},
+					"hosts": []string{
+						host + "/prometheus",
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply cluster/hooks.go]: Little bit too late for that, we did the same with Grafana, so this should be a global decision, but this is rather a UX issue, so I would involve someone else in it.

**Pull request merged**